packages/common-all/src/util/treeUtil.ts

```diff
@@ -90,26 +91,30 @@ export class TreeUtils {
     } else if (note.stub) {
       icon = TreeMenuNodeIcon.plusOutlined;
     }
+    const fm = PublishUtils.getPublishFM(note);
 
     return {
       key: note.id,
       title: note.title,
       icon,
       hasTitleNumberOutlined: note.fname.startsWith(TAGS_HIERARCHY),
       vaultName: VaultUtils.getName(note.vault),
-      navExclude: note.custom?.nav_exclude,
-      children: this.sortNotesAtLevel({
-        noteIds: note.children,
-        noteDict,
-        reverse: note.custom?.sort_order === "reverse",
-      })
-        .map((noteId) =>
-          TreeUtils.note2Tree({
-            noteId,
-            noteDict,
-          })
-        )
-        .filter(isNotUndefined),
+      navExclude: fm.nav_exclude || false,
+      children:
+        fm.nav_children_exclude || fm.has_collection
```

Comment:
nitpick: This part is getting pretty unreadable, could we refactor it to be cleaner?